src/NServiceBus.AcceptanceTests/NServiceBusAcceptanceTest.cs

```diff
@@ -36,5 +40,37 @@ public void SetUp()
                 return testName + "." + endpointBuilder;
             };
         }
+
+        [TearDown]
+        public void TearDown()
+        {
+            if (!TestExecutionContext.CurrentContext.TryGetRunDescriptor(out var runDescriptor))
+            {
+                return;
+            }
+
+            var scenarioContext = runDescriptor.ScenarioContext;
+
+            if (Environment.GetEnvironmentVariable("CI") != "true" || Environment.GetEnvironmentVariable("VERBOSE_TEST_LOGGING")?.ToLower() == "true")
+            {
+                TestContext.WriteLine($@"Test settings:
+{string.Join(Environment.NewLine, runDescriptor.Settings.Select(setting => $"   {setting.Key}: {setting.Value}"))}");
+
+                TestContext.WriteLine($@"Context:
+{string.Join(Environment.NewLine, scenarioContext.GetType().GetProperties().Select(p => $"{p.Name} = {p.GetValue(scenarioContext, null)}"))}");
```

Comment:
all of this could also just be dumped via `ScenarioContext.AddTrace` in the testing framework but then the current behavior of logging this information regardless of the test outcome would change.